src/cbc_sdk/platform/audit.py

```diff
@@ -39,3 +67,275 @@ def get_auditlogs(cb):
         """
         res = cb.get_object("/integrationServices/v3/auditlogs")
         return res.get("notifications", [])
+
+
+"""Query Class"""
+
+
+class AuditLogRecordQuery(BaseQuery, QueryBuilderSupportMixin, CriteriaBuilderSupportMixin,
+                          ExclusionBuilderSupportMixin, IterableQueryMixin, AsyncQueryMixin):
+    """
+    Query object that is used to locate ``AuditLogRecord`` objects.
+
+    The ``AuditLogRecordQuery`` is constructed via SDK functions like the ``select()`` method on ``CBCloudAPI``.
+    The user would then add a query and/or criteria to it before iterating over the results.
+
+    The following criteria may be added to the query via the standard ``add_criteria()`` method, or added to query
+    exclusions via the standard ``add_exclusions()`` method:
+
+    * ``actor_ip`` - IP address of the entity that caused the creation of this audit log.
+    * ``actor`` - Name of the entity that caused the creation of this audit log.
+    * ``request_url`` - URL of the request that caused the creation of this audit log.
+    * ``description`` - Text description of this audit log.
+    """
+
+    def __init__(self, doc_class, cb):
+        """
+        Initialize the ``AuditLogRecordQuery``.
+
+        Args:
+            doc_class (class): The model class that will be returned by this query.
+            cb (BaseAPI): Reference to API object used to communicate with the server.
+        """
+        self._doc_class = doc_class
+        self._cb = cb
+        self._count_valid = False
+        super(AuditLogRecordQuery, self).__init__()
+
+        self._query_builder = QueryBuilder()
+        self._criteria = {}
+        self._exclusions = {}
+        self._sortcriteria = {}
+        self._search_after = None
+        self.num_remaining = None
+        self.num_found = None
+        self.max_rows = -1
+
+    @staticmethod
+    def _create_valid_time_filter(kwargs):
+        """
+        Verifies that an alert criteria key has the timerange functionality
```

Comment:
`alert criteria` ?

Reply:
Yeah, that's why this was still only a draft PR, because I wanted one more chance to go over all the docstrings. :)